utils/zerotrie/src/zerotrie.rs

```diff
@@ -628,24 +629,28 @@ impl_zerotrie_subtype!(
     ZeroTrieSimpleAscii,
     String,
     reader::get_iter_ascii_or_panic,
+    core::iter::Map<reader::ZeroTrieIterator<'_>, fn((Vec<u8>, usize)) -> (String, usize)>,
```

Comment:
Nit: I'm in support of changing the `impl Iterator` to a concrete type, but I don't want to restrict the concrete type to being a specific map composition like this. Please introduce a new type such as `ZeroTrieAsciiIterator` or something like that so we can change the internals in the future. Safest would be to have a different type for all four concrete impls.

Reply:
And these four types redirect all 76 iterator methods to the types they wrap?

Reply:
Fair point, we can clean this up later before zerotrie 1.0

Reply:
I've made a `#[doc(hidden)]` type alias, should be non-clickable in the docs